apache-maven/src/assembly/shared/validate.cmd

```diff
@@ -46,20 +46,21 @@ if exist "%USERPROFILE%\mavenrc_pre.cmd" call "%USERPROFILE%\mavenrc_pre.cmd" %*
 set ERROR_CODE=0
 
 @REM ==== START VALIDATION ====
-if not "%JAVA_HOME%"=="" goto OkJHome
+if not "%JAVA_HOME%"=="" goto javaHomeSet
 for %%i in (java.exe) do set "JAVACMD=%%~$PATH:i"
-goto checkJCmd
+goto checkJavaCmd
 
-:OkJHome
+:javaHomeSet
 set "JAVACMD=%JAVA_HOME%\bin\java.exe"
 
-:checkJCmd
 if not exist "%JAVACMD%" (
-  echo The JAVA_HOME environment variable is not defined correctly, >&2
-  echo this environment variable is needed to run this program. >&2
+  echo The JAVA_HOME environment variable is not defined correctly, so Apache Maven cannot be started. >&2
+  echo JAVA_HOME is set to "%JAVA_HOME%", but "%%JAVA_HOME%%\bin\java.exe" does not exist. >&2
   goto error
 )
 
-if "%JAVA_HOME%"=="" (
-  echo Warning: JAVA_HOME environment variable is not set. >&2
+:checkJavaCmd
+if not exist "%JAVACMD%" (
+  echo The java.exe command does not exist in PATH, so Apache Maven cannot be started. >&2
```

Comment:
I think it should be:
`The java.exe command does not exist in PATH nor JAVA_HOME is set, so Apache Maven cannot be started.`
At this point we should provide all options.

Reply:
This makes sense. Will change.